py-polars/polars/io/delta.py

```diff
@@ -313,3 +315,114 @@ def _check_if_delta_available() -> None:
         raise ImportError(
             "deltalake is not installed. Please run `pip install deltalake>=0.9.0`."
         )
+
+
+def _reconstruct_field_type(
```

Comment:
I believe this currently does not map `LargeString` types yet, does it?

It would be nice to have some dedicated tests for this method to show that it works as intended.

Reply:
Indeed, it skips LargeString because this is catched by the `DeltaTable.schema().to_pyarrow(large_types=True)`

Reply:
If we include it here though (LargeBinary/LargeString/LargeList), the logic becomes way easier to read! We can get rid of a whole bunch of code. And it will make more sense.

Reply:
> If we include it here though (LargeBinary/LargeString/LargeList mapping), the logic becomes way easier to read! We can get rid of a whole bunch of code. And it will make more sense.

Good point! I'll look into creating a helper function that can accommodate also these Large types

Reply:
I've added the large types to normal types mapping.